spec/example_app/app/controllers/docs_controller.rb

```diff
@@ -9,11 +16,23 @@ def index
   end
 
   def show
-    render_page "docs/#{params[:page]}"
+    render_correct_page
   end
 
   private
 
+  def find_special_file
+    SPECIAL_FILES.select { |page| page[:page] == params[:page] }.first
+  end
+
+  def render_correct_page
+    if !find_special_file.nil?
+      render_page find_special_file[:file]
```

Comment:
A tiny one: I have a preference to wrapping method call arguments with a `()` to make it explicit what it's doing:

```suggestion
      render_page(find_special_file[:file])
```

i.e.: here I now know that `find_special_file` is returning a hash of some kind and `render_page` is a method call